### PR TITLE
rose metadata: url: add section or ns root

### LINKED ISSUE #52: rose metadata: root web url
It may be useful to have an inherited url root that options append a particular piece to, rather than specifying site-specific urls each time.

### PR DESCRIPTION
**Pull request by benfitzpatrick**
This closes #52.

@arjclark, please review.
@oliver-sanders, please review 2.

### CODE REVIEW

File: doc/rose-configuration-metadata.html [hunk 1]
```diff
@@ -879,11 +881,13 @@ <h3 id="meta.conf.help">Metadata for Help</h3>
...
         <pre class="prettyprint lang-rose_conf">
-url=http://www-nwp/~frve/VER/view/dev/doc/GTDP5.html
+url=http://www.something.com/FOO/view/dev/doc/FOO.html
 </pre>
 
         <p>For example, the config editor will trigger a web browser to display
-        this when a variable name is clicked.</p>
+        this when a variable name is clicked. A partial URL can be used for
+        variables (e.g. <kbd>url=#foobar</kbd> if the variable's section or
+        namespace has a relevant parent url property to use as a prefix.</p>
```
Comment (arjclark) [inline doc/rose-configuration-metadata.html]: Could you pop an example in here of using the partial URL?

**Comment by arjclark** [comment]
Documentation addition request but otherwise all looks good and seems to be working based on manual testing of the cylc gui with the demo app.

**Comment by benfitzpatrick** [comment]
Documentation improved

**Comment by arjclark** [comment]
Looks OK to me. No problems from the test-battery in my environment.

Over to you @oliver-sanders

**Comment by oliver-sanders** [comment]
Tested as working, passes test-battery.

**Pull request merged**